### PR TITLE
backing: move the min votes threshold to the runtime

### PR DESCRIPTION
**Pull request by alindima**
Copy of https://github.com/paritytech/polkadot/pull/7577
Also addresses review comments.

For previous reviewers: I maintained the list of commits from previous PR. The only unreviewed commits, are the ones starting with: https://github.com/paritytech/polkadot-sdk/pull/1200/commits/e7e82bc11b50014f2886c87460b9225524e42d35, which address the review comments

### CODE REVIEW

File: polkadot/runtime/parachains/src/configuration.rs [hunk 1]
```diff
@@ -292,6 +295,7 @@ impl<BlockNumber: Default + From<u32>> Default for HostConfiguration<BlockNumber
 			on_demand_fee_variability: Perbill::from_percent(3),
 			on_demand_target_queue_utilization: Perbill::from_percent(25),
 			on_demand_ttl: 5u32.into(),
+			minimum_backing_votes: 2,
```
Comment (eskimor) [inline polkadot/runtime/parachains/src/configuration.rs:298]: Why not use the `LEGACY_MIN_BACKING_VOTES` constant here?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3468303

**Comment by alindima** [comment]
@sandreim @eskimor, could you please have another look? I addressed the review comments and I think that if it looks good, it could be merged.

Runtime upgrades are currently failing on Versi even with master code, so the reason is unrelated to this PR.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3494434

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3494434

**Comment by alindima** [comment]
> The CI pipeline was cancelled due to failure one of the required jobs. **Job name**: cargo-clippy **Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3494434

wut? the job passed..

File: cumulus/client/relay-chain-rpc-interface/src/rpc_client.rs [hunk 2]
```diff
@@ -588,6 +588,16 @@ impl RelayChainRpcClient {
 			.await
 	}
 
+	/// Get the minimum number of backing votes for a candidate.
+	pub async fn parachain_host_minimum_backing_votes(
```
Comment (eskimor) [inline cumulus/client/relay-chain-rpc-interface/src/rpc_client.rs:592]: Is this even needed in cumulus? It should not.

**Review by eskimor — APPROVED** [review]

File: cumulus/client/relay-chain-rpc-interface/src/rpc_client.rs (hunk 2 above)
Comment (alindima) [reply cumulus/client/relay-chain-rpc-interface/src/rpc_client.rs:592]: It doesn't compile without it, I see all runtime APIs have counterparts here.
I'm not sure why they're needed though

Comment (ordian) [reply cumulus/client/relay-chain-rpc-interface/src/rpc_client.rs:592]: I had similar question some time ago, and got this suggestion from @skunert: https://github.com/paritytech/cumulus/pull/2160#issuecomment-1562490741

File: polkadot/runtime/kusama/src/lib.rs [hunk 3]
```diff
@@ -1741,6 +1741,8 @@ pub mod migrations {
 
 		// Upgrade SessionKeys to include BEEFY key
 		UpgradeSessionKeys,
+
```
Comment (sandreim) [inline polkadot/runtime/kusama/src/lib.rs:1744]: nit: a comment describing what the migration does is useful.

**Review by sandreim — APPROVED** [review]

File: cumulus/client/relay-chain-rpc-interface/src/rpc_client.rs (hunk 2 above)
Comment (alindima) [reply cumulus/client/relay-chain-rpc-interface/src/rpc_client.rs:592]: I think we can open a separate issue to discuss this further

Comment (alindima) [reply cumulus/client/relay-chain-rpc-interface/src/rpc_client.rs:592]: see: https://github.com/paritytech/polkadot-sdk/issues/1332

**Pull request merged**